src/loaders/OMETiffLoader.js

```diff
@@ -0,0 +1,322 @@
+            );
+          }
+        }
+        const image = await tiff.getImage(pyramidIndex);
+        // Flips bits for us for endianness.
+        const raster = await image.readRasters({ pool: poolOrDecoder });
+        return raster[0];
+      })
+    );
+    // Get first selectioz * SizeZ * SizeT * SizeC + loaderSelection[0]n size as proxy for image size.
+    const image = await tiff.getImage(
+      z * SizeZ * SizeT * SizeC + loaderSelection[0]
+    );
+    const width = image.getWidth();
+    const height = image.getHeight();
+    return { data: rasters, width, height };
+  }
+
+  /**
+   * Returns image width and height (at pyramid level z) without fetching data.
+   * This information is inferrable from the provided omexml.
+   * This is only used by the OverviewLayer for inferring the box size.
+   * It is NOT the actual pixel-size but rather the image size
+   * without any padding.
+   * @param {number} z positive integer (0 === highest zoom level)
+   * @returns {Object} width: number, height: number
+   */
+  getRasterSize({ z }) {
+    const { width, height } = this;
+    /* eslint-disable no-bitwise */
+    return {
+      height: height >> z,
+      width: width >> z
+    };
+    /* eslint-disable no-bitwise */
+  }
+
+  async _getChannel({ image, x, y }) {
+    const tile = await image.getTileOrStrip(x, y, 0, this.poolOrDecoder);
+    const is8Bits = image.fileDirectory.BitsPerSample[0] === 8;
+    const is16Bits = image.fileDirectory.BitsPerSample[0] === 16;
+    const is32Bits = image.fileDirectory.BitsPerSample[0] === 32;
+    const data =
+      (is8Bits && new Uint8Array(tile.data)) ||
+      (is16Bits && new Uint16Array(tile.data)) ||
+      (is32Bits && new Uint32Array(tile.data));
+    // Javascript needs little endian byteorder, so we flip if the data is not.
+    // eslint-disable-next-line no-unused-expressions
+    if (!image.littleEndian) {
+      flipEndianness(data);
+    }
+    return data;
+  }
+
+  /**
+   * Converts Array of loader selection objects into an IFD index.
+   *
+   * Ex.
+   *  const loaderSelectionObj = [
+   *     { time: 1, channel: 'a', z: 0 },
+   *     { time: 1, channel: 'b', z: 0 }
+   *  ];
+   *  const serialized = loader.serializeSelection(loaderSelectionObj);
+   *  console.log(serialized);
+   *  // 4, 5
+   *
+   * @param {Array || Object} loaderSelectionObjs Human-interpretable array of desired selection objects
+   * @returns {Array} number[][], IFD indices for selecting an image of the tiff.
+   */
+  serializeSelection(loaderSelectionObjs) {
+    // Wrap selection in array if only one is provided
+    const selectionObjs = Array.isArray(loaderSelectionObjs)
+      ? loaderSelectionObjs
+      : [loaderSelectionObjs];
+    const serialized = selectionObjs.map(obj => this._serialize(obj));
+    return serialized;
+  }
+
+  _serialize(selectionObj) {
+    const serializedSelection = this._getIFDIndex(selectionObj);
+    return serializedSelection;
+  }
```

Comment:
My comment before is whether you think we could move this within the `getTile` and `getRaster` calls? I think it should be easy with zarr and it would make managing state really simple.

Reply:
Yes, I agree.